centreon/www/class/centreonMonitoring.class.php

```diff
@@ -76,61 +85,92 @@ public function getPoller()
+     * @return int
      */
-    public function getServiceStatusCount($host_name, $objXMLBG, $o, $status, $obj)
-    {
-            $rq = "SELECT count(distinct s.service_id) as count, 1 AS REALTIME "
-                . "FROM services s, hosts h " . (!$objXMLBG->is_admin ? ", centreon_acl " : "")
-                . "WHERE s.state = '" . $status . "' "
-                . "AND s.host_id = h.host_id "
-                . "AND s.enabled = '1' "
-                . "AND h.enabled = '1' "
-                . "AND h.name = '" . $host_name . "' ";
-
-            # Acknowledgement filter
-        if ($o == "svcSum_ack_0") {
-            $rq .= "AND s.acknowledged = 0 AND s.state != 0 ";
-        } elseif ($o == "svcSum_ack_1") {
-            $rq .= "AND s.acknowledged = 1 AND s.state != 0 ";
+    public function getServiceStatusCount(
+        string $hostName,
+        CentreonXMLBGRequest $centreonXMLBGRequest,
+        string $o,
+        int $serviceStatus
+    ): int {
+        $toBind = [':service_status' => $serviceStatus, ':host_name' => $hostName];
+        if ($centreonXMLBGRequest->is_admin) {
+            $query = <<<SQL
+                SELECT count(distinct s.service_id) as count, 1 AS REALTIME
+                FROM services s
+                INNER JOIN hosts h
+                    ON h.host_id = s.host_id
+                WHERE s.state = :service_status
+                    AND s.host_id = h.host_id
+                    AND s.enabled = '1'
+                    AND h.enabled = '1'
+                    AND h.name = :host_name
+                SQL;
+        } else {
+            $accessGroups = $centreonXMLBGRequest->access->getAccessGroups();
+            [$bindValues, $subRequest] = $this->createMultipleBindQuery(array_keys($accessGroups), ':grp_');
+            $query = <<<SQL
+                SELECT count(distinct s.service_id) as count, 1 AS REALTIME
+                FROM services s
+                INNER JOIN hosts h
+                    ON h.host_id = s.host_id
+                INNER JOIN centreon_acl acl
+                    ON acl.host_id = h.host_id
+                    AND acl.service_id = s.service_id
+                WHERE s.state = :service_status
+                    AND s.host_id = h.host_id
+                    AND s.enabled = '1'
+                    AND h.enabled = '1'
+                    AND h.name = :host_name
+                    AND acl.group_id IN ({$subRequest})
+                SQL;
+            $toBind = [...$toBind, ...$bindValues];
         }
 
-        if (!$objXMLBG->is_admin) {
-            $rq .=  "AND h.host_id = centreon_acl.host_id "
-                . "AND s.service_id = centreon_acl.service_id "
-                . "AND centreon_acl.group_id IN (" .  $obj->access->getAccessGroupsString() . ") ";
+        # Acknowledgement filter
+        if ($o === 'svcSum_ack_0') {
+            $query .= ' AND s.acknowledged = 0 AND s.state != 0 ';
+        } elseif ($o === "svcSum_ack_1") {
```

Comment:
You can use constants no ?

Reply:
Yes, as long as you take the time to understand the function as a whole. Which I don't want to do at the moment.

Reply:
I understand, it's better to have good names for the constants :D T resolve.